kn/fotos/api.py

```diff
@@ -123,18 +129,16 @@ def _set_metadata(data, request):
         rotation = data['rotation']
         if rotation not in [0, 90, 180, 270]:
             return {'error': 'rotation is not valid'}
+        entity.set_rotation(rotation, save=False)
 
-    result = {'Ok': True}
+        result['largeSize'] = entity.get_cache_size('large')
 
-    entity.update_visibility([visibility])
     entity.set_title(title, save=False)
-    if isinstance(entity, fEs.FotoEntity):
-        entity.set_description(description, save=False)
-        entity.set_rotation(rotation, save=False)
-        result['largeSize'] = entity.get_cache_size('large')
-        result['thumbnailSize'] = entity.get_cache_size('thumb')
-    # save all changes (except for visibility) in one batch
+
+    # save changes in one batch
     entity.save()
+    # except for visibility which is much harder to save in the same batch
+    entity.update_visibility([visibility])
```

Comment:
Ja. Hij slaat hem twee keer op. Ik zou hem kunnen herschrijven dat hij het in dezelfde batch doet, maar dat is naar mijn mening onredelijk veel moeite. Dit werkt, is snel (want hoe vaak verander je de visibility van een album?) en is leesbaar.
Het wordt alleen twee keer opgeslagen als de visibility daadwerkelijk veranderd, en de extra tijd die daar in gaat valt toch in het niet met het recursief updaten van de `effectiveVisibility`.